### PR TITLE
Adds StreetAddress2, default device type, and "other" to account request template

### LINKED ISSUE #1277: Add Street address 2 and default device type params
Our goal is to fully automated user registration process. In order to do the information we collect in the sign up form must match what is needed to create an organization. The signup form currently asks for Unit type and Unit number this needs to migrate to Street address 2. The sign up form doesn't ask for a default device type and needs too

TODO

- Add an optional field Street Address 2 to the account-request params
- Add an optional field device type to the account-request params

### PR DESCRIPTION
**Pull request by emmastephenson**
## Related Issue or Background Info

Fixes #1277 

## Changes Proposed

Change the backend account-request email template to include:
- street address 2 instead of unit
- default device type
- "other" categories for facility type and device type

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java [hunk 1]
```diff
@@ -61,17 +62,19 @@ public String getTemplateName() {
...
     variableMap.put("facilityType", facilityType);
+    variableMap.put("facilityTypeOther", facilityTypeOther);
     variableMap.put("organizationName", organizationName);
     variableMap.put("facilityName", facilityName);
     variableMap.put("cliaNumber", cliaNumber);
     variableMap.put("testingDevices", testingDevices);
+    variableMap.put("testingDevicesOther", testingDevicesOther);
+    variableMap.put("defaultTestDevice", defaultTestDevice);
     variableMap.put("accessDevices", accessDevices);
     variableMap.put("browsers", browsers);
```
Comment (maya) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java:79]: We also have an ["other browser" new field](https://github.com/CDCgov/prime-simplereport-site/pull/103/files#diff-67012e167bb9599500bd2ec11dad8bce9556b42c94ff5d6cc8d97154a052532eR254). Could you add that too?

**Review by maya — CHANGES_REQUESTED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java (hunk 1 above)
Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java:79]: Yep! I'll add it.

File: backend/src/main/resources/templates/account-request.html [hunk 2]
```diff
@@ -26,6 +26,7 @@ <h1>Testing facility information</h1>
 <b>Default testing device: </b>[[${defaultTestDevice}]]<br>
 <b>Devices accessing SimpleReport: </b>[[${accessDevices}]]<br>
 <b>Web browsers: </b>[[${browsers}]]<br>
+<b>Other Web browsers: </b>[[${browsersOther}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:29]: Can you update to "Sentence casing"? (Our preferred style on the project)

File: backend/src/main/resources/templates/account-request.html [hunk 3]
```diff
@@ -11,19 +11,22 @@ <h1>Facility administrator</h1>
...
+<b>Facility Type Other: </b>[[${facilityTypeOther}]]<br>
 <b>Organization name: </b>[[${organizationName}]]<br>
 <b>Testing facility name: </b>[[${facilityName}]]<br>
 <b>Clinical Laboratory Improvement Amendments (CLIA) number: </b>[[${cliaNumber}]]<br>
 <b>Testing devices: </b>[[${testingDevices}]]<br>
+<b>Other testing devices: </b>[[${testingDevicesOther}]]<br>
+<b>Default testing device: </b>[[${defaultTestDevice}]]<br>
 <b>Devices accessing SimpleReport: </b>[[${accessDevices}]]<br>
 <b>Web browsers: </b>[[${browsers}]]<br>
+<b>Other Web browsers: </b>[[${browsersOther}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:29]:
```suggestion
<b>Other web browsers: </b>[[${browsersOther}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 4]
```diff
@@ -11,19 +11,22 @@ <h1>Facility administrator</h1>
...
 <b>Street Address: </b>[[${mailingAddress1}]]<br>
-<b>Unit Type: </b>[[${aptSuiteOther}]]<br>
-<b>Unit number: </b>[[${aptFloorSuiteNo}]]<br>
+<b>Street Address 2: </b>[[${streetAddress2}]]<br>
 <b>City: </b>[[${city}]]<br>
 <b>State: </b>[[${state}]]<br>
 <b>ZIP code: </b>[[${zip}]]<br>
 <b>County: </b>[[${county}]]<br>
 <b>Facility Type: </b>[[${facilityType}]]<br>
+<b>Facility Type Other: </b>[[${facilityTypeOther}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:20]:
```suggestion
<b>Other facility type: </b>[[${facilityTypeOther}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 5]
```diff
@@ -11,19 +11,22 @@ <h1>Facility administrator</h1>
 <h1>Testing facility information</h1>
 
 <b>Street Address: </b>[[${mailingAddress1}]]<br>
-<b>Unit Type: </b>[[${aptSuiteOther}]]<br>
-<b>Unit number: </b>[[${aptFloorSuiteNo}]]<br>
+<b>Street Address 2: </b>[[${streetAddress2}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:14]:
```suggestion
<b>Street address 2: </b>[[${streetAddress2}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 6]
```diff
@@ -11,19 +11,22 @@ <h1>Facility administrator</h1>
 <h1>Testing facility information</h1>
 
 <b>Street Address: </b>[[${mailingAddress1}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:13]:
```suggestion
<b>Street address: </b>[[${mailingAddress1}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 7]
```diff
@@ -36,8 +39,7 @@ <h1>Ordering provider</h1>
 <b>National Provider Identifier (NPI): </b>[[${npi}]]<br>
 <b>Phone number: </b>[[${opPhoneNumber}]]<br>
 <b>Street address: </b>[[${opMailingAddress1}]]<br>
-<b>Unit type: </b>[[${opAptSuiteOther}]]<br>
-<b>Unit number: </b>[[${opAptFloorSuiteNo}]]<br>
+<b>Street address 2: </b>[[${opStreetAddress2}]]<br>
 <b>City: </b>[[${opCity}]]<br>
 <b>State: </b>[[${opState}]]<br>
 <b>Zip Code: </b>[[${opZip}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:45]:
```suggestion
<b>Zip code: </b>[[${opZip}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 8]
```diff
@@ -11,19 +11,22 @@ <h1>Facility administrator</h1>
...
 
 <b>Street Address: </b>[[${mailingAddress1}]]<br>
-<b>Unit Type: </b>[[${aptSuiteOther}]]<br>
-<b>Unit number: </b>[[${aptFloorSuiteNo}]]<br>
+<b>Street address 2: </b>[[${streetAddress2}]]<br>
 <b>City: </b>[[${city}]]<br>
 <b>State: </b>[[${state}]]<br>
 <b>ZIP code: </b>[[${zip}]]<br>
 <b>County: </b>[[${county}]]<br>
 <b>Facility Type: </b>[[${facilityType}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:19]:
```suggestion
<b>Facility type: </b>[[${facilityType}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 9]
```diff
@@ -10,20 +10,23 @@ <h1>Facility administrator</h1>
...
 <b>County: </b>[[${county}]]<br>
-<b>Facility Type: </b>[[${facilityType}]]<br>
+<b>Facility type: </b>[[${facilityType}]]<br>
+<b>Other facility type: </b>[[${facilityTypeOther}]]<br>
 <b>Organization name: </b>[[${organizationName}]]<br>
 <b>Testing facility name: </b>[[${facilityName}]]<br>
 <b>Clinical Laboratory Improvement Amendments (CLIA) number: </b>[[${cliaNumber}]]<br>
 <b>Testing devices: </b>[[${testingDevices}]]<br>
+<b>Other testing devices: </b>[[${testingDevicesOther}]]<br>
+<b>Default testing device: </b>[[${defaultTestDevice}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:26]:
```suggestion
<b>Primary testing device: </b>[[${defaultTestDevice}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 10]
```diff
@@ -10,20 +10,23 @@ <h1>Facility administrator</h1>
...
 <b>ZIP code: </b>[[${zip}]]<br>
 <b>County: </b>[[${county}]]<br>
-<b>Facility Type: </b>[[${facilityType}]]<br>
+<b>Facility type: </b>[[${facilityType}]]<br>
+<b>Other facility type: </b>[[${facilityTypeOther}]]<br>
 <b>Organization name: </b>[[${organizationName}]]<br>
 <b>Testing facility name: </b>[[${facilityName}]]<br>
 <b>Clinical Laboratory Improvement Amendments (CLIA) number: </b>[[${cliaNumber}]]<br>
 <b>Testing devices: </b>[[${testingDevices}]]<br>
+<b>Other testing devices: </b>[[${testingDevicesOther}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:25]:
```suggestion
<b>Other testing device: </b>[[${testingDevicesOther}]]<br>
```

File: backend/src/main/resources/templates/account-request.html [hunk 11]
```diff
@@ -10,20 +10,23 @@ <h1>Facility administrator</h1>
...
+<b>Default testing device: </b>[[${defaultTestDevice}]]<br>
 <b>Devices accessing SimpleReport: </b>[[${accessDevices}]]<br>
 <b>Web browsers: </b>[[${browsers}]]<br>
+<b>Other web browsers: </b>[[${browsersOther}]]<br>
```
Comment (maya) [inline backend/src/main/resources/templates/account-request.html:29]:
```suggestion
<b>Other web browser: </b>[[${browsersOther}]]<br>
```

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java [hunk 12]
```diff
@@ -17,19 +17,22 @@
   @NotNull private String workPhoneNumber;
   private String cellPhoneNumber;
   @NotNull private String mailingAddress1;
-  private String aptSuiteOther;
-  private String aptFloorSuiteNo;
+  private String streetAddress2;
```
Comment (brmaden) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java:20]: Is there a reason we use `mailingAddress1` and `streetAddress2`?  It seems like we should settle on either mailing or street.  These aren't stored, but if they were would probably use street.

Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java:20]: No particular reason - I just wasn't sure if it used mailingAddress1 for consistency elsewhere. Changed to use streetAddress1 and streetAddress2 instead.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java [hunk 13]
```diff
@@ -16,20 +16,23 @@
   @NotNull private String email;
   @NotNull private String workPhoneNumber;
   private String cellPhoneNumber;
-  @NotNull private String mailingAddress1;
-  private String aptSuiteOther;
-  private String aptFloorSuiteNo;
+  @NotNull private String streetAddress1;
+  private String streetAddress2;
```
Comment (brmaden) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java:20]: Renaming existing fields that are `@NotNull` will break requests from the deployed site.  We can use Jackson to handle this, though.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java [hunk 14]
```diff
@@ -133,28 +137,20 @@ public void setCellPhoneNumber(String cellPhoneNumber) {
     this.cellPhoneNumber = cellPhoneNumber;
   }
 
-  public String getMailingAddress1() {
-    return mailingAddress1;
+  public String getstreetAddress1() {
+    return streetAddress1;
```
Comment (brmaden) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java:141]: Would expect `public String getStreetAddress1()` for this and the setter.   However, instead, I suggest we use Jackson and remove all of these getters/setters (except `getEmail()` which is used elsewhere).

**Review by brmaden — CHANGES_REQUESTED** [review]
There are some change here that would require us to stay in sync with `prime-simplereport-site`.  One way we can mitigate is by using Jackson more (and this has the side effect of letting us remove most of the getters and setters).

I suggest we use `@JsonProperty` on the fields we expect as input.  Jackson's `@JsonAlias` can be used to allow the alternate name `mailing-address1` for the field that has been renamed (allowing it to be accepted in the request as either `street-address1` or `mailing-address1`).  (We shouldn't have to worry about both being specified in the same request, but if that happened then the one that appears last in the request body is used)

**Review by maya — APPROVED** [review]
Approving my previous requested changes ✅

**Review by brmaden — APPROVED** [review]

**Pull request merged**